### PR TITLE
Assignment1

### PR DESCRIPTION
**Pull request by prajjwalkumarpanzade**
Assignment 1 completed

### CODE REVIEW

File: Assignment1/ip.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+puts "Enter IP address "       #Print message.
+IP_add = gets.chomp            #Input from user.
+split = IP_add.split('.',0)    #Split input on .
+cls = split[0].to_i            #convert 0 index of splited array to integer
+if (1..127) === cls            #Checking between classes
```
Comment (samu80) [inline Assignment1/ip.rb:5]: Your approach is also correct, but you can use case statements like this:
```
case cls
when 1..127
  puts "Class A"
when 128..191
  puts "Class B"
when 192..223
  puts "Class C"
when 224..239
  puts "Class D"
when 240..254
  puts "Class E"
end
```

File: Assignment1/q1.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+puts "Enter number"
+
+number = gets.to_i
+flag = 0
+if number > 1
+  for i in 2 ... number do
+
+   if number % i == 0
+    flag = 1
+
```
Comment (samu80) [inline Assignment1/q1.rb]: Once you get the divisible number, you can break the loop; there is no need to check further as the number is not prime.

File: Assignment1/q1.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#1. WAP to display number is prime or not
+
+puts "Enter number"
+
+number = gets.to_i
+flag = 0
```
Comment (samu80) [inline Assignment1/q1.rb]: You can use boolean values here like:
flag = false

File: Assignment1/q1.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+if number > 1
+  for i in 2 ... number do
+
+   if number % i == 0
+    flag = 1
+
+   end
+  end
+
+  if flag == 0
```
Comment (samu80) [inline Assignment1/q1.rb]: Use ternary operator:
flag == 0 ? "Number is prime" : "Number is not prime"

File: Assignment1/q2.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+#if grade is between 1-5 return elementary
+#if grade is between 6-8 return middle school
+#if grade is between 9-12 return high school
+#if grade is between otherwise return college
+
+
+puts "Enter grade"
+grade = gets.to_i
+
+if (1..5) === grade
```
Comment (samu80) [inline Assignment1/q2.rb:9]: Here also you can use case and when statement

File: Assignment1/q4.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
+mystr = "Hello I am Prajjwalkumar Panzade my email id is prajjwalbpanzade22@gmail.com, Gender - M, amount = Rs.120000,my contact number = 9112422777"
+
+name = name = mystr.match(/[A-Z]\w+ [A-Z]\w+/)[0]
```
Comment (samu80) [inline Assignment1/q4.rb]: Why this "name" field is repeated

File: Assignment1/q3.rb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+arr = Array("a".."z")
+puts arr.reverse
```
Comment (samu80) [inline Assignment1/q3.rb]: Please write the loop logic for reversing the array

File: Assignment1/q4.rb (hunk 6 above)
Comment (prajjwalkumarpanzade) [reply Assignment1/q4.rb]: By mistake

File: Assignment1/q1.rb [hunk 8]
```diff
@@ -0,0 +1,19 @@
+puts "Enter number"
+
+number = gets.to_i
+flag = 0
+if number > 1
+  for i in 2 ... number do
+
+   if number % i == 0
+    flag = 1
```
Comment (samu80) [inline Assignment1/q1.rb]: Here flag is not required, you can remove it and can check i == number ? "Prime" : "Not Prime"

Comment (samu80) [inline Assignment1/q1.rb]: Here flag is not required, you can remove it and can check i == number ? "Prime" : "Not Prime"

File: Assignment1/ip.rb [hunk 9]
```diff
@@ -0,0 +1,21 @@
+puts "Enter IP address "       #Print message.
+IP_add = gets.chomp            #Input from user.
+split = IP_add.split('.',0)    #Split input on .
+cls = split[0].to_i            #convert 0 index of splited array to integer
+if (1..127) === cls            #Checking between classes 
+  puts "Class A"
+
+elsif (128..191) === cls
+  puts "Class B"
+
```
Comment (samu80) [inline Assignment1/ip.rb]: Please do not add unnecessary blank lines

**Review by samu80 — APPROVED** [review]

**Pull request merged**